process/track/blockProcessor.go

```diff
@@ -453,7 +455,8 @@ func (bp *blockProcessor) requestHeadersIfNothingNewIsReceived(
 	}
 
 	shouldRequestHeaders := bp.roundHandler.Index()-int64(highestRoundInReceivedHeaders) > process.MaxRoundsWithoutNewBlockReceived &&
-		int64(latestValidHeader.GetNonce())-int64(lastNotarizedHeaderNonce) <= process.MaxHeadersToRequestInAdvance
+		int64(latestValidHeader.GetNonce())-int64(lastNotarizedHeaderNonce) <= process.MaxHeadersToRequestInAdvance &&
+		!bp.isImportDBMode // TODO: consider updating this condition to work for import db as well
```

Comment:
why is an importDB mode needed?

Reply:
the problem is with the legacy condition.

The condition does not capture the fact that when in importDB the node would only receive headers it requests, while the round index is correct, so the difference between any received header and the round index will always cause this condition to be true.

In that case the node will consider that it needs to request headers here, when in fact in importdb it already received the requested blocks.